File: DSA/CP_And_Contests/CodeForces/Div-4_21-07-2023/TenWordsOfWisdom.c
#include<stdio.h>

int main()
{
	int t;
	
	scanf("%d",&t);

	int count = 0;

	while(count!=t)
	{
		int n;
		
		scanf("%d",&n);

		int max = 1;

		int cnt = 0;
		
		int res,i=0;
		
		while(cnt!=n)
		{
			i++;
			
			int a,b;
			
			scanf("%d %d",&a,&b);

			if(a<=10)
			{
				if(b>max)
				{
					max = b;
					
					res = i;
				}
			}
			
			cnt++;
		}
		
		printf("%d\n",res);
	}
	
	return 0;
}

File: DSA/CP_And_Contests/CodeForces/Div-4_21-07-2023/TenWordsOfWisdom.py
t = int(input())

count = 0

while count!=t:

	n = int(input())

	maximum = 1

	cnt = 0
	
	i=0
	res=0
	
	while(cnt!=n):
	
		i+=1
		
		a,b = map(int, input().split())

		if a<=10:
		
			if b>maximum:
			
				maximum = b
				
				res = i
		
		cnt+=1
	
	print(res)